api/Controllers/HotelsController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using iTechArt.Hotels.Api.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace iTechArt.Hotels.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class HotelsController : Controller
+    {
+        private readonly HotelsDatabaseContext _hotelsDb;
+
+        public HotelsController(HotelsDatabaseContext hotelsDb)
+        {
+            _hotelsDb = hotelsDb;
+        }
+
+        [HttpPost]
+        [Authorize(Roles = "admin")]
+        public async Task<IActionResult> CreateHotel([FromBody] Hotel request )
+        {
+            if (request == null)
+            {
+                return BadRequest("Wrong response");
+            }
+
+            Hotel hotel = new Hotel
+            {
+                Name = request.Name,
+                Country = request.Country,
+                Sity = request.Sity,
+                Address = request.Address
+            };
```

Comment:
As I can see, you've received `Hotel` instance and then just copied all the fields for exactly the same type? Why do you need it? Is it possible to use the same instance for `Add` method without creating new one?

Reply:
The right question should be - why do you even use Hotel as a model in action?

Reply:
That was my next question :)

Reply:
When I create new hotel I should give all hotel information. That is the reason I use database hotel model